lib/rtcpeerconnection/edge.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+var EventTarget = require('../util/eventtarget');
+var inherits = require('util').inherits;
+var util = require('../util');
+
+var match = navigator.userAgent.match(/Edge\/(\d+).(\d+)$/);
+var PeerConnection = require('rtcpeerconnection-shim')(window, match[2]);
+
+function EdgeRTCPeerConnection(configuration) {
+  if (!(this instanceof EdgeRTCPeerConnection)) {
+    return new EdgeRTCPeerConnection(configuration);
+  }
+
+  EventTarget.call(this);
+
+  var peerConnection = new PeerConnection(configuration);
+
+  Object.defineProperties(this, {
+    _peerConnection: {
+      value: peerConnection
+    },
+    iceConnectionState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.iceConnectionState;
+      }
+    },
+    iceGatheringState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.iceGatheringState;
+      }
+    },
+    signalingState: {
+      enumerable: true,
+      get: function() {
+        return this._peerConnection.signalingState;
+      }
+    }
+  });
+
+  util.proxyProperties(PeerConnection.prototype, this, peerConnection);
+}
+
+inherits(EdgeRTCPeerConnection, EventTarget);
+
+// otalk/RTCPeerConnection shim throws InvalidAccessError
+// when sender is not part of getSenders or has been previously removed
+EdgeRTCPeerConnection.prototype.removeTrack = function removeTrack(sender) {
+  if (!this._peerConnection._isClosed && (sender instanceof window.RTCRtpSender)) {
+    var s = this._peerConnection.getSenders().find(function(s) {
+      return s === sender;
+    });
+    if (!s) {
+      return;
+    }
+  }
+  this._peerConnection.removeTrack(sender);
```

Comment:
Just wondering: would it be better to rewrite this method as

```js
EdgeRTCPeerConnection.prototype.removeTrack = function removeTrack(sender) {
  try {
    this._peerConnection.removeTrack(sender);
  } catch (error) {
    if (error.message.startsWith('InvalidAccessError') {
      return;
    }
    throw error;
  }
};
```

Feels just a bit simpler to maintain? WDYT @manjeshbhargav